releasenotes/notes/commutation-checker-gates-15349cfa13b40e50.yaml

```diff
@@ -0,0 +1,20 @@
+---
+features_circuits:
+  - |
+    The :class:`.CommutationChecker` class has a new argument
+    called `gates` in the initializer, which allows to compute the
+    commutations only on a subset of gates. This enables speedups
+    in cases where not all commutations are of interest.
+
+    Example::
+
+        from qiskit.transpiler.passes import CommutationAnalysis
+        from qiskit.circuit.commutation_library import CommutationChecker, StandardGateCommutations
+
+        gates = ["cx", "x", "rx"]
+        commutation_checker = CommutationChecker(
+            StandardGateCommutations, gates=gates
+        )
+
+        # run commutation analysis only on cx, x and rx gates
+        analysis = CommutationAnalysis(commutation_checker)
```

Comment:
I don't think we need to document this, we don't seem to include `CommutationChecker` in the docs anywhere so it's not a public class.

Reply:
Removed in https://github.com/Qiskit/qiskit/commit/351d5f11958e23d0b489b8138de0cb48a0e6a3c8